test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -387,6 +422,98 @@ func runWorkload(b *testing.B, tc *testCase, w *workload) []DataItem {
 				mu.Unlock()
 			}
 
+		case *churnOp:
+			var namespace string
+			if concreteOp.Namespace != nil {
+				namespace = *concreteOp.Namespace
+			} else {
+				namespace = fmt.Sprintf("namespace-%d", opIndex)
+			}
+
+			var churnFns []func(name string) string
+
+			// Node.
+			if concreteOp.NodeTemplatePath != nil {
+				node := &v1.Node{}
+				if err := getSpecFromFile(concreteOp.NodeTemplatePath, node); err != nil {
+					b.Fatalf("op %d: unable to parse Node: %v", opIndex, err)
+				}
+				churnFns = append(churnFns, func(name string) string {
+					if name != "" {
+						clientset.CoreV1().Nodes().Delete(ctx, name, metav1.DeleteOptions{})
+						return ""
+					}
+
+					n, err := clientset.CoreV1().Nodes().Create(ctx, node, metav1.CreateOptions{})
+					if err != nil {
+						return ""
+					}
+					return n.Name
+				})
+			}
+			// Pod.
+			if concreteOp.PodTemplatePath != nil {
+				pod := &v1.Pod{}
+				if err := getSpecFromFile(concreteOp.PodTemplatePath, pod); err != nil {
+					b.Fatalf("op %d: unable to parse Node: %v", opIndex, err)
+				}
+				churnFns = append(churnFns, func(name string) string {
+					if name != "" {
+						clientset.CoreV1().Pods(namespace).Delete(ctx, name, metav1.DeleteOptions{})
+						return ""
+					}
+
+					p, err := clientset.CoreV1().Pods(namespace).Create(ctx, pod, metav1.CreateOptions{})
+					if err != nil {
+						return ""
+					}
+					return p.Name
+				})
+			}
+			// Service.
+			if concreteOp.ServiceTemplatePath != nil {
+				svc := &v1.Service{}
+				if err := getSpecFromFile(concreteOp.ServiceTemplatePath, svc); err != nil {
+					b.Fatalf("op %d: unable to parse Service: %v", opIndex, err)
+				}
+				churnFns = append(churnFns, func(name string) string {
+					if name != "" {
+						clientset.CoreV1().Services(namespace).Delete(ctx, name, metav1.DeleteOptions{})
+						return ""
+					}
+
+					s, err := clientset.CoreV1().Services(namespace).Create(ctx, svc, metav1.CreateOptions{})
+					if err != nil {
+						return ""
+					}
+					return s.Name
+				})
+			}
+
+			if len(churnFns) == 0 {
+				b.Fatalf("op %d: at least one churn function needs to be defined", opIndex)
```

Comment:
we should do this in isValid (check if at least one of {Node,Pod,Service}TemplatePath is nil). doing this in isValid will let the framework exit early instead of at workloda runtime